File: src/main/java/com/gmail/borlandlp/minigamesdtools/localizer/text/TextNode.java
package com.gmail.borlandlp.minigamesdtools.localizer.text;

import java.util.Set;

/**
 * Содержит в себе представление какой-либо строки на различные языки.
 */
public interface TextNode {
    /**
     * @return Список доступных языков для этой ноды
     */
    Set<String> containsLanguages();

    /**
     * @param langCode - двухзначный код языка, для которого будет запрошена строка ноды.
     * @return String неформатированное представление ноды для langCode
     */
    String getForLanguage(String langCode);
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/localizer/text/TextNode.kt
package com.gmail.borlandlp.minigamesdtools.localizer.text

/**
 * Содержит в себе представление какой-либо строки на различные языки.
 */
interface TextNode {
    /**
     * @return Список доступных языков для этой ноды
     */
    fun containsLanguages(): Set<String>

    /**
     * @param langCode - двухзначный код языка, для которого будет запрошена строка ноды.
     * @return String неформатированное представление ноды для langCode
     */
    fun getForLanguage(langCode: String): String
}